src/storage/mod.rs

```diff
@@ -1,13 +1,34 @@
 use std::collections::HashMap;
 
+use thiserror::Error;
 use tokio::sync::mpsc;
 use tokio::sync::oneshot;
 
-use crate::types::{Key, Value};
+use crate::types::{Blob, Key, Value};
 
 pub enum StorageCommand {
     Set(Key, Value),
     Get(Key),
+    Incr(Key),
+    Decr(Key),
+}
+
+#[derive(Error, Debug)]
+pub enum StorageError {
+    #[error("integer overflow")]
+    Overflow,
+
+    #[error("not an integer")]
+    NotAnInteger,
+
+    #[error("unknown reason")]
+    Failed(std::io::Error),
+}
+
+impl From<std::io::Error> for StorageError {
+    fn from(err: std::io::Error) -> StorageError {
+        StorageError::Failed(err)
+    }
 }
```

Comment:
`thiserror` can automatically create the `impl From` here with this syntax:

```rust
#[error("unknown reason")]
Failed(#[from] std::io::Error),
```

You can also use the enum value in the message if you want: `"unknown reason: {0}"`.

Reply:
Oooh that's much nicer. Thank you! I made that change in de1bea1.